content/guides/getstarted/solana-test-validator.md

```diff
@@ -0,0 +1,342 @@
+with the `--reset` flag:
+
+```shell
+solana-test-validator --reset
+```
+
+If the ledger exists, this command will reset the ledger to genesis, which
+resets the state by deleting all existing data and starting fresh.
+
+## Runtime Features
+
+Solana has a feature set mechanism that allows you to enable or disable certain
+blockchain features when running the test validator. By default, the test
+validator runs with all runtime features activated.
+
+To query the runtime feature status:
+
+```shell
+solana feature status <ADDRESS>
+```
+
+- `ADDRESS` is the feature status to query [default: all known features]
+
+To activate a specific feature:
+
+```shell
+solana feature activate <FEATURE_KEYPAIR> <CLUSTER>
+```
+
+- `FEATURE_KEYPAIR` is the signer for the feature to activate
+- `CLUSTER` is the cluster to activate the feature on
+
+To deactivate specific features in genesis:
+
+```shell
+solana-test-validator --deactivate-feature <FEATURE_PUBKEY>
+```
+
+## Changing Versions
+
+To check your current `solana-test-validator` version:
+
+```shell
+solana-test-validator --version
+```
+
+Your `solana-test-validator` runs on the same version as the Solana CLI version.
+
+To test your programs against different versions of the Solana runtime, you can
+install multiple versions of the Solana CLI and switch between them using the
+solana-install set command:
+
+```shell
+solana-install init <VERSION>
+```
+
+- `VERSION` is the desired CLI version to install
+
+Make sure to restart your Solana test validator after changing versions to
+ensure it runs the correct version.
+
+## Cloning Programs
+
+To add existing on-chain programs to your local environment, you can clone the
+program.
+
+To clone an account from the cluster:
+
+```shell
+solana-test-validator --clone PROGRAM_ADDRESS
+```
+
+This is useful for testing interactions with standard programs.
+
+To clone an upgradeable program and its executable data from the cluster:
+
+```shell
+solana-test-validator --clone-upgradeable-program PROGRAM_ADDRESS
+```
+
+## Resetting State on Accounts at Startup
+
+By default the validator will resume an existing ledger _(if present)_. But
+during startup, you can reset the ledger either to genesis or to specific
+account state that you provide.
+
+### Reset to Genesis
+
+To reset the ledger to the genesis state:
+
+```shell
+solana-test-validator --reset
+```
+
+### Reset to Specific Accounts
+
+To reset the state of specific accounts every time you start the validator, you
+can use a combination of account snapshots and the `--account` flag.
+
+First, save the desired state of an account as a JSON file:
+
+```shell
+solana account PROGRAM_ADDRESS --output json > account_state.json
+```
+
+Then load this state each time you reset the validator:
+
+```shell
+solana-test-validator --reset --account PROGRAM_ADDRESS account_state.json
+```
+
+## Interacting with a Running Test Validator
```

Comment:
Imo this section should come first after installation. Get the developer doing useful stuff asap, then teach them how to properly configure everything later.